core/pom.xml

```diff
@@ -234,83 +206,37 @@
         <dependency>
             <groupId>com.google.cloud</groupId>
             <artifactId>google-cloud-bigquery</artifactId>
-            <version>1.48.0</version>
         </dependency>
         <dependency>
             <groupId>com.google.cloud</groupId>
             <artifactId>google-cloud-nio</artifactId>
-            <version>0.83.0-alpha</version>
         </dependency>
         <dependency>
             <groupId>org.apache.flink</groupId>
             <artifactId>flink-clients_2.11</artifactId>
             <version>1.5.5</version>
         </dependency>
 
-        <!-- Jackson due to jinjava dependency problems -->
```

Comment:
This suggests these were declared here only for the sake of jinjava removed below, so with jinjava removed I tried removing these too and nothing broke. The Jackson versions are set in `<dependencyManagement>` so if they're brought in transitively and the versions are important, that should take care of it already.

Reply:
@davidheryanto @zhilingc do you have context on these?

I feel like we can just go ahead and remove these.

Reply:
For a bit of extra confidence, I've run `mvn dependency:analyze -DignoreNonCompile` and indeed Jackson is not used in `core` even transitively, nor are Jinjava or Mustache. The dependency plugin's analyzer isn't perfect but it's pretty good outside of annotations.

Pushing another commit that removes more of the reported unused deps with no apparent breakage.

Reply:
Yes, the jinjava library was used as a templating engine for our SQL query in Core to retrieve batch features. But in 0.3, batch feature retrieval responsibility is moved to Serving, so we don't need need jinjava anymore in Core